hydra-node/src/Hydra/Network/Reliability.hs

```diff
@@ -222,28 +234,28 @@ withReliability tracer me otherParties withRawNetwork callback action = do
     let mknownAckForUs = knownAcks !? myIndex
     case (mmessageAckForUs, mknownAckForUs) of
       (Just messageAckForUs, Just knownAckForUs) ->
-         -- We resend messages if our peer notified us that it's lagging behind our
-         -- latest message sent
-         when (messageAckForUs < knownAckForUs) $ do
-           let missing = fromList [messageAckForUs + 1 .. knownAckForUs]
-           messages <- readTVarIO sentMessages
-           forM_ missing $ \idx -> do
-             case messages IMap.!? idx of
-               Nothing ->
-                 traceWith tracer $
-                   ReliabilityFailedToFindMsg $
-                     "FIXME: this should never happen, there's no sent message at index "
```

Comment:
Suggestion: Fix the `FIXME` :) We should have a proper structured message instead of building a string manually here.